pkg/pool-weighted/contracts/lbp/LBPool.sol

```diff
@@ -254,18 +264,21 @@ contract LBPool is ILBPool, WeightedPool, Ownable2Step, BaseHooks {
     /// @inheritdoc ILBPool
     function getLBPoolImmutableData() external view override returns (LBPoolImmutableData memory data) {
         data.tokens = _vault.getPoolTokens(address(this));
+        data.projectTokenIndex = _projectTokenIndex;
+        data.reserveTokenIndex = _reserveTokenIndex;
+
         (data.decimalScalingFactors, ) = _vault.getPoolTokenRates(address(this));
         data.isProjectTokenSwapInBlocked = _blockProjectTokenSwapsIn;
         data.startTime = _startTime;
         data.endTime = _endTime;
 
         data.startWeights = new uint256[](_TWO_TOKENS);
-        data.startWeights[_projectTokenIndex] = _projectTokenStartWeight;
-        data.startWeights[_reserveTokenIndex] = _reserveTokenStartWeight;
+        data.startWeights[data.projectTokenIndex] = _projectTokenStartWeight;
+        data.startWeights[data.reserveTokenIndex] = _reserveTokenStartWeight;
 
         data.endWeights = new uint256[](_TWO_TOKENS);
-        data.endWeights[_projectTokenIndex] = _projectTokenEndWeight;
-        data.endWeights[_reserveTokenIndex] = _reserveTokenEndWeight;
+        data.endWeights[data.projectTokenIndex] = _projectTokenEndWeight;
+        data.endWeights[data.reserveTokenIndex] = _reserveTokenEndWeight;
```

Comment:
This is not storage, so I don't think it makes much of a difference to use `_projectTokenIndex` and `_reserveTokenIndex`. Either way should be fine.

Reply:
Yeah; on second thought, I think I will change it back to use the immutables directly. There should be no performance difference (just switching one compile-time constant for another), but using the immutable indexes directly is clearer.